### DISCUSSION TITLE
correct mime type for SRT transcripts?

### DISCUSSION THREAD

**Discussion by daveajones** [Podcastindex-org/podcast-namespace#595] [Clarification Needed]
@tedhosmann mentioned that `text/srt` might be the actual mime type for SRT transcript format, but the more I investigate, the more confused I get:

- https://en.wikipedia.org/wiki/SubRip - Wikipedia indicates that `application/x-subrip` is correct, but then contradicts itself in the footnotes but doesn't give a replacement.
- https://developer.wordpress.org/reference/hooks/mime_types/ - Wordpress defines it as `text/plain`, which I've also seen expressed in other places

I can't find `text/srt` documented anywhere yet.

**Comment by ryan-lp** [comment]
`text/srt` is definitely not the actual mime type for SRT.

SRT does not have an official MIME type [assigned by the IANA](https://www.iana.org/assignments/media-types/media-types.xhtml) and nor is there a single standard for the format in its modern form as it is actually used today. Despite this, it is the dominant subtitles format today, and the quasi set of standards has evolved over the decades into a set of features and MIME types from which we can survey what MIME types are actually widely adopted. The more commonly used MIME types are: `application/x-subrip`, `text/plain` and less commonly `application/octet-stream`.

`text/srt` is not a thing.

What I would say is that interoperability is important, so although there is no IANA-recognised standard MIME type, we should be doing something that is compatible with what others are doing.

The Apache MIME types file gives `application/x-subrip`:

https://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types

The Android source code gives `application/x-subrip`:

https://android.googlesource.com/platform/frameworks/base/+/3c864beb5a15/mime/java-res/android.mime.types

I believe from memory, Android can also accept `text/plain`. However, in practice, an App that edits or opens SRT files and wants to filter its Open File dialog by files with the correct MIME type, `application/x-subrip` is obviously preferred as it will filter exactly the types of files you want, whereas `text/plain` is too general and will list all text files. Conversely, if you are clicking on a file in a web page served by Apache, and you want to open up that file in the relevant app, `application/x-subrip` is a more appropriate MIME type than `text/plain` since it will filter exactly the open-with apps that are relevant for that type, rather than any generic app.

**Reply by jamescridland** [reply]
Ryan - in your opinion, does SRT do anything better than VTT?

Given we have clear preference by Apple for VTT, should the spec be opinionated and recommend VTT over SRT in the same way?

“It is recommended to produce a JSON and VTT format file; JSON for high fidelity word timings, and VTT for compatibility with Apple and browser-based playback. An SRT may be used as a fallback to VTT.”

**Reply by daveajones** [reply]
@jamescridland My understanding is that VTT was an expanded version of SRT adding stylesheets and other things.  It's basically SRT+.  So, by definition, SRT is less capable.  WebVTT was actually originally called WebSRT.

**Reply by ryan-lp** [reply]
Not quite a strict superset. VTT changed the syntax of the timestamps so that an SRT file is not a valid VTT file. Although in practice SRT/VTT parsers tend to be bilingual.

That said, many of the most widely used transcription tools export only to SRT and not VTT (e.g. Otter.ai) due to the longer foothold that SRT has had historically. That would be the reality at this point in time, so I don't think the technical superiority of one format over another is the deciding factor, and I see no reason to make the spec opinionated on the matter if Apple accepts both VTT and SRT as a fallback.

The JSON question I think is a separate matter. For example, Buzzsprout does not agree to reserving the JSON format for word timings, that format was also never assigned its own MIME type and filename extension, so there are a few issues there that are a separate discussion.

**Reply by daveajones** [reply]
Instead of being opinionated it might be best to just list which apps support which transcript types and leave at that. The individual app devs can submit PR’s to clarify their supported formats. Many 2.0 apps use Transcriptator so will probably have broad support.

**Comment by daveajones** [comment]
Apple updated their docs to accommodate “text/srt” and “application/x-subrip” both, for SRT transcripts:

https://help.apple.com/itc/podcasts_connect/#/itcb54353390

I think we can close this one.

**Reply by tedhosmann** [reply]
Look again.  Thanks to everyone for the feedback shared here.

Only "application/srt" and  "application/x-subrip" should be used for SRT. I don't think this will create any new work for this format.

And "text/vtt" will be supported for VTT. This also shouldn't create new work.
